distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1966,14 +1952,105 @@ private CometDClient setupCometDClient(List<String> channels) throws Exception {
         cometDClient.start();
-        channels.stream()
-                .forEach(cometDClient::subscribe);
+        channels.forEach(cometDClient::subscribe);
         return cometDClient;
     }
 
+    private void setupParallelInterruptDownloadTest(String metacardId, String filename,
+            String resourceData) throws Exception {
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename), new ChunkedContent.ChunkedContentBuilder(
+                        resourceData).delayBetweenChunks(Duration.ofMillis(50))
+                        .build());
+    }
+
+    /**
+     * Default settings.
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2) throws Exception {
+        runParallelInterruptDownloadTest(client1,
+                client2,
+                DEFAULT_PARALLEL_CLIENT_START_DELAY,
+                false);
+    }
+
+    /**
+     * Default settings with choice of start order flag.
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2, boolean flipStartOrder) throws Exception {
+        runParallelInterruptDownloadTest(client1,
+                client2,
+                DEFAULT_PARALLEL_CLIENT_START_DELAY,
+                flipStartOrder);
+    }
+
+    /**
+     * EXECUTION LOGIC FOR PARALLEL INTERRUPTION OF DOWNLOADS
+     * In this template:
+     * We always start client2 first, unless the flipStartOrder flag is set
+     * We always interrupt client1
+     * We always validate the output integrity of client2
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2, long startDelay, boolean flipStartOrder) throws Exception {
+        String filename = "product1.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        setupParallelInterruptDownloadTest(metacardId, filename, resourceData);
+
+        if (flipStartOrder) {
+            client1.startDownload(metacardId);
+            Thread.sleep(startDelay);
+            client2.startDownload(metacardId);
+        } else {
+            client2.startDownload(metacardId);
+            Thread.sleep(startDelay);
+            client1.startDownload(metacardId);
+        }
+
+        Thread.sleep(600);
```

Comment:
Thread sleeps generally lead to unreliable tests in the long run as well. I would also like to see these replaced with concrete waits until an expected state is reached.

If that is not possible, I think we should at least be using a constant instead of these magic numbers.